plugin-java/che-plugin-java-ext-jdt/che-jdt-ext-machine/src/main/java/org/eclipse/ltk/internal/ui/refactoring/TextEditChangePreviewViewer.java

```diff
@@ -0,0 +1,262 @@
+		}
+	}
+//
+//	private static class ComparePreviewer extends CompareViewerSwitchingPane {
+//		private CompareConfiguration fCompareConfiguration;
+//		private String               fLabel;
+//		private ImageDescriptor      fDescriptor;
+//		private Image                fImage;
+//
+//		public ComparePreviewer(Composite parent) {
+//			super(parent, SWT.BORDER | SWT.FLAT, true);
+//			fCompareConfiguration = new CompareConfiguration();
+//			fCompareConfiguration.setLeftEditable(false);
+//			fCompareConfiguration.setLeftLabel(RefactoringUIMessages.ComparePreviewer_original_source);
+//			fCompareConfiguration.setRightEditable(false);
+//			fCompareConfiguration.setRightLabel(RefactoringUIMessages.ComparePreviewer_refactored_source);
+//			addDisposeListener(new DisposeListener() {
+//				public void widgetDisposed(DisposeEvent e) {
+//					if (fImage != null && !fImage.isDisposed())
+//						fImage.dispose();
+//				}
+//			});
+//			Dialog.applyDialogFont(this);
+//		}
+//
+//		public void setLabel(String label) {
+//			fLabel = label;
+//		}
+//
+//		public void setImageDescriptor(ImageDescriptor imageDescriptor) {
+//			fDescriptor = imageDescriptor;
+//		}
+//
+//		protected Viewer getViewer(Viewer oldViewer, Object input) {
+//			return CompareUI.findContentViewer(oldViewer, (ICompareInput)input, this, fCompareConfiguration);
+//		}
+//
+//		public void setText(String text) {
+//			if (fLabel != null) {
+//				super.setText(fLabel);
+//			} else {
+//				super.setText(text);
+//			}
+//			Image current = null;
+//			if (fDescriptor != null) {
+//				current = fImage;
+//				fImage = fDescriptor.createImage();
+//			} else {
+//				current = fImage;
+//				fImage = null;
+//			}
+//			setImage(fImage);
+//			if (current != null) {
+//				current.dispose();
+//			}
+//		}
+//	}
+//
+//	private static class CompareElement implements ITypedElement, IEncodedStreamContentAccessor, IResourceProvider {
+//		// we use an encoding that preserves Unicode across the stream
+//		private static final String ENCODING = "UTF-8";    //$NON-NLS-1$
+//		private String    fContent;
+//		private String    fType;
+//		private IResource fResource;
+//
+//		public CompareElement(String content, String type, IResource resource) {
+//			fContent = content;
+//			fType = type;
+//			fResource = resource;
+//		}
+//
+//		public String getName() {
+//			return RefactoringUIMessages.ComparePreviewer_element_name;
+//		}
+//
+//		public Image getImage() {
+//			return null;
+//		}
+//
+//		public String getType() {
+//			return fType;
+//		}
+//
+//		public InputStream getContents() throws CoreException {
+//			try {
+//				return new ByteArrayInputStream(fContent.getBytes(ENCODING));
+//			} catch (UnsupportedEncodingException e) {
+//				return new ByteArrayInputStream(fContent.getBytes());
+//			}
+//		}
+//
+//		public String getCharset() {
+//			return ENCODING;
+//		}
+//
+//		public IResource getResource() {
+//			return fResource;
+//		}
+//	}
+
```

Comment:
This is code from Eclipse. I do not want to refactor eclipse code. It can create problems when we will update to new Eclipse version.